WordPress/Classes/ViewRelated/Gutenberg/Utils/GutenbergMediaEditorImage.swift

```diff
@@ -0,0 +1,65 @@
+import Foundation
+
+/**
+ This is a struct to be given to MediaEditor that represent the image.
+ We need the full high-quality image in the Media Editor.
+ */
+class GutenbergMediaEditorImage: AsyncImage {
+    private var tasks: [URLSessionDataTask] = []
+
+    private var originalURL: URL
+
+    private var fullQualityURL: URL? {
+        guard var urlComponents = URLComponents(url: originalURL, resolvingAgainstBaseURL: false) else {
+            return nil
+        }
+
+        urlComponents.query = nil
```

Comment:
This will only work for WPCom.
On Self-Hosted, sized urls can be of this kind: `https://testingsiteetoledom.mystagingwebsite.com/wp-content/uploads/2019/11/beautiful-blur-bright-326055-1-150x150.jpg`.

I'm updating some Gutenberg code so we can get the original URL directly from the bridge delegate 👍

Reply:
That's good to know!

Thanks @etoledom!